Tests/XMLCoderTests/RJITest.swift

```diff
@@ -141,18 +141,68 @@ class RJITest: XCTestCase {
         let type: String
     }
 
-    func testRSS() throws {
-        let decoder = XMLDecoder()
-//        let encoder = XMLEncoder()
+    private struct NodeCodingStrategyProvider {
+        typealias Strategy = (CodingKey) -> XMLEncoder.NodeEncoding
+
+        private func strategy(for _: RSS.Type) -> Strategy {
+            return { codingKey in
+                print(codingKey.stringValue)
+                switch codingKey as! RSS.CodingKeys {
+                case .dc, .sy, .admin, .rdf, .content:
+                    return .attribute
+                case _:
+                    return .element
+                }
+            }
+        }
+
+        private func strategy(for _: Channel.GeneratorAgentKeys.Type) -> Strategy {
+            return { codingKey in
+                switch codingKey as! Channel.GeneratorAgentKeys {
+                case .resource:
+                    return .attribute
+                }
+            }
+        }
+
+        func strategy(for type: Any.Type) -> Strategy {
+            switch type {
+            case let concreteType as RSS.Type:
+                return strategy(for: concreteType)
+            case let concreteType as Channel.GeneratorAgentKeys.Type:
+                return strategy(for: concreteType)
+            case _:
+                return { _ in .default }
+            }
+        }
+    }
 
-        let dateFormatter = DateFormatter()
-        dateFormatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ssZZZZZ"
-        decoder.dateDecodingStrategy = .formatted(dateFormatter)
+    func dateFormatter() -> DateFormatter {
+        let formatter = DateFormatter()
+        formatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ssZZZZZ"
+        return formatter
+    }
 
+    func testRSS() throws {
```

Comment:
I plan to re-visit this test to enable it once all issues related to it have been resolved.